pennylane/decomposition/symbolic_decomposition.py

```diff
@@ -209,9 +210,9 @@ def _resource_fn(
         gate_counts[resource_rep(qml.PauliX)] = num_zero_control_values * 2
         return gate_counts
 
-    # pylint: disable=too-many-arguments
+    # pylint: disable=protected-access,too-many-arguments
     @register_condition(_condition_fn)
-    @register_resources(_resource_fn)
+    @register_resources(_resource_fn, work_wires=base_decomposition._work_wire_spec)
```

Comment:
Should we make a public `work_wire_spec_fn`?

Reply:
The private `_work_wire_spec` could be a function or just a dictionary, similar to how the resources could be a dictionary or a function. The public `work_wire_spec` is always a function.

Reply:
I have rename the public one to `get_work_wire_spec`